service/src/registry.ts

```diff
@@ -735,6 +739,36 @@ export class Registry {
     /*Codes_SRS_NODE_IOTHUB_REGISTRY_05_013: [The fromSharedAccessSignature method shall return a new instance of the `Registry` object.]*/
     return new Registry(config);
   }
+
+  private _normalizeAuthentication(deviceInfo: Registry.DeviceDescription): void {
+    if (!deviceInfo.hasOwnProperty('authentication')) {
+      /* Codes_SRS_NODE_IOTHUB_REGISTRY_06_028: [A device information with no authentication will be normalize with the following authentication:
+      authentication : {
+        type: 'sas',
+        symmetricKey: {
+          primaryKey: '',
+          secondaryKey: ''
+        }
+      }
+      ] */
+      deviceInfo.authentication = {
+          type: 'sas',
+          symmetricKey: {
+            primaryKey: '',
+            secondaryKey: ''
+          }
+        };
+    /* Codes_SRS_NODE_IOTHUB_REGISTRY_06_029: [** A device information with an authentication object that contains a `type` property is considered normalized.] */
+    } else if (!deviceInfo.authentication.hasOwnProperty('type')) {
```

Comment:
My reply is:

1) You prefaced your comment with nitpick.  I view nitpick as something of no great importance as viewed by the author of the comments.

2) The negative predicates are there because that is how the decisions (at least in my mind) needed to make this code work arise.  Therefore to me it is perfectly reasonable to write the code that way.

3) In this particular case these are EXACTLY the correct questions to ask with precisely the correct sequence.


---
In reply to: [128658002](https://github.com/Azure/azure-iot-sdk-node/pull/61#discussion_r128658002) [](ancestors = 128658002)